### DISCUSSION TITLE
questions about package size

### DISCUSSION THREAD

**Discussion by MoonShiesty** [aptos-labs/aptos-developer-discussions#81] [Questions] [answered]
our code deployment is close to the package size limit. is there a way to estimate the deployment payload size without deploying the contract?

ive deployed our contracts on testnet [here](https://explorer.aptoslabs.com/account/0x576bbf3df2119fdd557ede2e1c79e01243db39c1535051b94c407da33acefa2c/modules/code/market?network=testnet) and the bytecode size of the package appears to exceeds 64kb. is the payload compressed, or has `max_transaction_size_in_bytes` changed to support payloads greater than 64kb?

**Comment by Alexzander-Stone** [comment, marked as answer]
Revising your questions into smaller chunks:

#### 1. Can one retrieve the package size without publishing a package?
Yes, you can find the `package size` when running `aptos move publish` on the Aptos CLI.
The output will include a line with a structure of `package size {BYTES} bytes`.

Adding the Aptos CLI option `--profile-gas` is a quick way to simulate this action locally. Feel free to read more about this command by using the `--help` CLI option for `aptos move publish`.

Here's an example of the output I received when running `aptos move publish --profile-gas`:
```
package size 1017 bytes
Simulating transaction locally with the gas profiler...
{
  "Result": {
    // ...
  }
}
```

#### 2. What is the `max_transaction_size_in_bytes`?
[`max_transaction_size_in_bytes` is 64kb](https://github.com/aptos-labs/aptos-core/blame/main/aptos-move/aptos-gas-schedule/src/gas_schedule/transaction.rs#L68).

#### 3. Are payloads compressed when publishing?
Yes. 
Additionally, this causes the resulting size of the cached payload on the Aptos Explorer to not be equivalent to the compressed payload during the publishing of a package. If you'd like to know your published package size, please refer to the answer from question 1.

**Comment by MoonShiesty** [comment]
awesome this is all super helpful.

i noticed that `--profile-gas` doesn't appear to be working for `aptos move create-resource-account-and-publish-package` and there's no `--gas-profile` flag

**Reply by Alexzander-Stone** [reply]
Thanks for pointing that out, updated the option text to `--profile-gas`. `--gas-profile` was incorrect.

Are you seeing an error when `--profile-gas` is used for `create-resource-account-and-publish-package`?
